R/gg-plots.R

```diff
@@ -347,18 +347,23 @@ ggally_cor <- function(
         y <- as.numeric(y)
       }
 
-      corObj <- stats::cor.test(x, y, method = method, use = use)
-
-      # make sure all values have X-many decimal places
-      cor_est <- as.numeric(corObj$estimate)
-      cor_txt <- formatC(cor_est, digits = digits, format = "f")
-
-      # if stars should be added
-      if (isTRUE(stars)) {
-        cor_txt <- str_c(
-          cor_txt,
-          signif_stars(corObj$p.value)
-        )
+      if (length(x) <= 2 | length(y) <= 2) {
+        warning("Less than 2 observations, returning NA")
+        cor_txt <- NA
+      } else {
+        corObj <- stats::cor.test(x, y, method = method, use = use)
+
+        # make sure all values have X-many decimal places
+        cor_est <- as.numeric(corObj$estimate)
+        cor_txt <- formatC(cor_est, digits = digits, format = "f")
+
+        # if stars should be added
+        if (isTRUE(stars)) {
+          cor_txt <- str_c(
+            cor_txt,
+            signif_stars(corObj$p.value)
+          )
+        }
```

Comment:
Taking the approach from https://github.com/ggobi/ggally/pull/513/files#diff-7af1e69ba80a0f5423ab07b42e989a8bd7c2e79575c9f8e54244bb17a2a2a075R352

```suggestion
      if (length(x) < 3 | length(y) < 3) {
        warning("Less than 2 observations, returning NA")
        return("NA")
      }

      corObj <- stats::cor.test(x, y, method = method, use = use)

      # make sure all values have X-many decimal places
      cor_est <- as.numeric(corObj$estimate)
      cor_txt <- formatC(cor_est, digits = digits, format = "f")

      # if stars should be added
      if (isTRUE(stars)) {
        cor_txt <- str_c(
          cor_txt,
          signif_stars(corObj$p.value)
        )
```